earth2mip/beta/perturbation/bv.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+from typing import Callable
+
+import numpy as np
+import torch
+
+
+class BredVector:
+    """Bred Vector perturbation method, a classical technique for pertubations in
+    ensemble forecasting.
+
+    Parameters
+    ----------
+    model : Callable[[torch.Tensor], torch.Tensor]
+        Dynamical model, typically this is the prognostic AI model.
+        TODO: Update to prognostic looper
```

Comment:
Is it worth debating about where to put the model? In the constructor or in the `__call__` function?

Reply:
I think that keeping it in the `__init__` makes the `__call__` parsimonious for all of the other perturbation methods that do not require a model.

Reply:
Figured it would better to try to keep the perturbation call similar to other components. Also then its up to the user to load up the model they want for bred vector not us which I feel like could be useful. :shrug: